IfIsNotInA.java

```diff
@@ -0,0 +1,35 @@
+
+
+
+
+            /*21.	Given two array lists A and B, A has an element that is not in B while the other
+             elements are the same in both two lists. Find that missing element in B.
+              (This is a live coding challenge. Any programming language is fine.)*/
+
+
+import java.util.*;
+
+public class IfIsNotInA {
+    public static List solution(int[] first, int[] second) {
+        ArrayList<Integer> firstArrayList = new ArrayList<Integer>();
+        ArrayList<Integer> secondArrayList = new ArrayList<Integer>();
+        for (int i = 0; i < first.length; i++) {
+            firstArrayList.add(first[i]);
+        }
+        for (int i = 0; i < second.length; i++) {
+            secondArrayList.add(second[i]);
+        }
+
+        for (Integer x : secondArrayList)
+            if (firstArrayList.contains(x)) {
```

Comment:
`.contains(_)` called on an **array** is expensive. `O(n)`
Try using a `HashSet` (ideally yields `O(1)` but often in reality has a large constant in complexity) or a `SortedSet` that yield `O(log(n))`.